Dark Mode: update colors for star rating view

jaclync opened the pull request:
Stars UI for #1555 

## Changes

- Removed unused code `NoteTableViewCell` and xib that has the star rating UI
- Updated the colors of a star rating view so that the image color responds to Dark/Light mode changes by setting `UIImage` + `tintColor` of `UIImageView`. The star rating UI is used in these 3 cases:
  - Product Details
  - Review cell
  - Review Details

## Testing

- Launch the app
- Go to Products tab
- Tap on a Product with a 2-4 star --> the stars should have the expected color in both Light/Dark modes
- Go to Reviews tab --> the stars should have the expected color in both Light/Dark modes
- Tap on a Review with a 2-4 star --> the stars should have the expected color in both Light/Dark modes

## Example screenshots

Screen | Dark | Light
-- | -- | --
Product Details | ![Simulator Screen Shot - iPhone 8 - 2019-12-09 at 18 39 23](https://user-images.githubusercontent.com/1945542/70429408-70abd900-1ab3-11ea-877c-37a3f21ffa2f.png) | ![Simulator Screen Shot - iPhone 8 - 2019-12-09 at 18 39 28](https://user-images.githubusercontent.com/1945542/70429409-70abd900-1ab3-11ea-8f02-e0ad923b2038.png)
Review cell | ![Simulator Screen Shot - iPhone 8 - 2019-12-09 at 18 39 41](https://user-images.githubusercontent.com/1945542/70429411-70abd900-1ab3-11ea-8980-a4953e567b8a.png) | ![Simulator Screen Shot - iPhone 8 - 2019-12-09 at 18 39 47](https://user-images.githubusercontent.com/1945542/70429412-71446f80-1ab3-11ea-87fa-123801680625.png)
Review Details | ![Simulator Screen Shot - iPhone 8 - 2019-12-10 at 09 57 14](https://user-images.githubusercontent.com/1945542/70488386-7ea13e80-1b33-11ea-94f2-459a4fbf79ac.png) | ![Simulator Screen Shot - iPhone 8 - 2019-12-10 at 09 57 05](https://user-images.githubusercontent.com/1945542/70488384-7ea13e80-1b33-11ea-910e-a3b0a1ba0f00.png)










Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

jaclync committed:
Remove unused code.

jaclync committed:
Set image + tintColor for `UIImageView` to respond to Light/Dark mode changes.

mindgraffiti reviewed (commented):
Can the star ratings in the Review Details screen have the colors reversed? [Conversation and example](https://github.com/woocommerce/woocommerce-ios/pull/1573#issuecomment-562602856)

Stars in Product Details ✅ 
Stars in Reviews tab ✅

jaclync committed:
Create rating star filled/empty colors for usage in review details cell.

jaclync commented:
> Can the star ratings in the Review Details screen have the colors reversed?

updated here: https://github.com/woocommerce/woocommerce-ios/pull/1582/commits/f9ae623e37c8e379011389d94a8d006b2608b644 and the screenshots are also updated
